plugins/modules/pnp_intent.py

```diff
@@ -793,16 +825,50 @@ class instance for further use.
                 params=planned_count_params,
             )
 
+            dev_details_response = self.dnac_apply['exec'](
+                family="device_onboarding_pnp",
+                function="get_device_by_id",
+                params={"id": self.have["device_id"]}
+            )
+
+            pnp_state = dev_details_response.get("deviceInfo").get("state")
+
             if not self.want["site_name"]:
                 self.result['response'] = self.have.get("device_found")
                 self.result['msg'] = "Device is already added"
             else:
+                update_payload = {"deviceInfo": self.want.get('pnp_params')[0].get("deviceInfo")}
+                update_response = self.dnac_apply['exec'](
+                    family="device_onboarding_pnp",
+                    function="update_device",
+                    params={"id": self.have["device_id"],
+                            "payload": update_payload},
+                    op_modifies=True,
+                )
+                self.log(str(update_response))
+
+                if pnp_state == "Error":
+                    reset_paramters = self.get_reset_params()
+                    reset_response = self.dnac_apply['exec'](
+                        family="device_onboarding_pnp",
+                        function="update_device",
+                        params={"payload": reset_paramters},
+                        op_modifies=True,
+                    )
+                    self.log(str(reset_response))
+                    self.result['msg'] = "Device reset done Successfully"
+                    self.result['response'] = reset_response
+                    self.result['diff'] = self.validated_config
+                    self.result['changed'] = True
+
```

Comment:
Do we need to return here? or we need to print "Device is already claimed" ?

Reply:
I am returning here since, reset takes time. Hence, anyone using it should first reset and then claim